go/private/mode.bzl

```diff
@@ -210,6 +224,9 @@ def link_mode_args(mode):
     elif mode.link == LINKMODE_PLUGIN:
         if platform in _LINK_PLUGIN_PLATFORMS:
             args.append("-dynlink")
+    elif mode.link == LINKMODE_PIE:
+        if platform in _LINK_PIE_PLATFORMS:
+            args.append("-shared")
```

Comment:
Based on the output of `go build -x -work -buildmode=pie` with a cgo binary, we need to:

* Build the standard library with `-shared` (done, I think).
* Pass `-shared` to the compiler. (done).
* Pass `-pie` to C compiler when building cgo binary (done).
* Pass `-shared` to assembler (done).
* Pass `-buildmode=pie` to linker (done).

So this should be good.